### ISSUE TITLE
Repost Warning does not refresh when adding a # to the end of a link

### ISSUE THREAD

**Issue by allthefoxes** [honestbleeps/Reddit-Enhancement-Suite#2048]
A sneaky way to get past a report warning is to add # to the end of a link. While changing the link in any way does update the checker, it seems like adding this # to the end does not

**Comment by honestbleeps** [comment]
this is a reddit limitation that I'm not sure RES can really get around... unfortunately some sites use hash / hashbang for actual navigation, especially "web application" type sites... 

I'm not going to close this, because someone will probably chime in explaining a situation where I'm wrong, which I'm entirely open to... just sharing my first reaction/thought :)

**Comment by allthefoxes** [comment]
I just mean that, can't RES detect that we have added the # to the link, 
and check to see if its been posted? Since without the # the warning 
will come up, with the #, it will

**Comment by honestbleeps** [comment]
it could, but the problem is that sometimes it's completely legitimate to add #foobar to a link, that's the tricky part.

a plain # at the end of a link has no real function, but is also not necessarily a nefarious attempt to get around the repost filter, because a lot of sites link to # to scroll to the top or load the "main" part of the app, etc... it's also nontrivial in javascript to remove that "#" in a cross-browser way, which is why RES didn't do it for so long (and the fact that it does do it now actually causes some back/forward nav issues, but I digress)...

basically, it's a tricky situation with no real great answer :-\

**Comment by allthefoxes** [comment]
No, I understand that par :/

What I am saying is that reddit treats http://example.org as 
http://example.org#thing as two different things, two different posts.

Example: In subreddit A, User A posts http://example.org

In subreddit A, User B goes to submit http://example.org#thing

User B will see, on the same page, RES's warning about 'That link was 
already submitted to Subreddit A", however, when he submits the link, he 
will not get the "Someone already posted that" warning, and will be able 
to submit just fine.

Are we on the same page here?

**Comment by allthefoxes** [comment]
I'm having issues reproducing anyways, doesn't matter.

**Comment by erikdesjardins** [comment]
Yeah, submitHelper intentionally [removes stuff from the link,](https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/4adc0408a93d2058c987693573b5c67ea0b5c59c/lib/modules/submitHelper.js#L77) including the stuff after `#`, so it gets the same results for `foo.com#bar` and `foo.com` - but it would be easy to stop it from doing that.

It does this because for some reason (I don't know what I was thinking) I didn't verify [this claim](https://www.reddit.com/r/Enhancement/comments/2ntdxu/feature_request_automatically_fill_in_previous/cmhdjjo)  - I guess it just seemed reasonable enough to believe...

**Comment by erikdesjardins** [comment]
@allthefoxes Are you sure? I get a warning when I try to do it (might have to retract my last comment):

![image](https://cloud.githubusercontent.com/assets/7673145/6364115/dc395ec4-bc6f-11e4-9f5b-952ea5bcfd78.png)

**Comment by allthefoxes** [comment]
I was able to repost a subreddit with a #, but perhaps I am wrong.

Sorry for the wild goose chase.
